packages/wrangler/src/cloudchamber/apply.ts

```diff
@@ -553,31 +610,69 @@ export async function applyCommand(
 		}
 
 		if (action.action === "modify") {
-			const [_result, err] = await wrap(
-				promiseSpinner(
-					ApplicationsService.modifyApplication(action.id, action.application),
-					{
-						json: args.json,
-						message: `modifying application ${action.name}`,
+			{
+				const [_result, err] = await wrap(
+					promiseSpinner(
+						ApplicationsService.modifyApplication(action.id, {
+							...action.application,
+							configuration: undefined,
+						})
+					)
+				);
+
+				if (err !== null) {
+					if (!(err instanceof ApiError)) {
+						crash(
+							`Unexpected error modifying application ${action.name}: ${err.message}`
+						);
 					}
-				)
-			);
-			if (err !== null) {
-				if (!(err instanceof ApiError)) {
+
+					if (err.status === 400) {
+						crash(
+							`Error modifying application ${action.name} due to a ${brandColor.underline("misconfiguration")}\n\n\t${formatError(err)}`
+						);
+					}
+
 					crash(
-						`Unexpected error modifying application ${action.name}: ${err.message}`
+						`Error modifying application ${action.name} due to an internal error (request id: ${err.body.request_id}): ${formatError(err)}`
 					);
 				}
+			}
+
+			if (action.rollout_step_percentage !== undefined) {
+				const [_result, err] = await wrap(
+					promiseSpinner(
+						RolloutsService.createApplicationRollout(action.id, {
+							description: "Progressive update",
+							strategy: CreateApplicationRolloutRequest.strategy.ROLLING,
+							target_configuration:
+								(action.application
+									.configuration as ModifyDeploymentV2RequestBody) ?? {},
+							step_percentage: action.rollout_step_percentage,
+						}),
+						{
+							json: args.json,
+							message: `rolling out container version ${action.name}`,
+						}
+					)
+				);
+				if (err !== null) {
+					if (!(err instanceof ApiError)) {
+						crash(
+							`Unexpected error rolling out application ${action.name}: ${err.message}`
+						);
+					}
+
+					if (err.status === 400) {
+						crash(
```

Comment:
```suggestion
						throw new UserError(
```

Reply:
these are all outdated now, you can check from latest commit :pray: